src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/CodeGenerationIntegrationTest.cs

```diff
@@ -178,7 +204,13 @@ public class CodeGenerationIntegrationTest(bool designTime = false)
     public void IncompleteTagHelper() => RunTagHelpersTest(TestTagHelperDescriptors.DefaultPAndInputTagHelperDescriptors);
 
     [IntegrationTestFact]
-    public void BasicTagHelpers() => RunTagHelpersTest(TestTagHelperDescriptors.DefaultPAndInputTagHelperDescriptors);
+    public void BasicTagHelpers()
+    {
+        // Trying to load the DLL results in "System.BadImageFormatException: Bad IL format."
```

Comment:
Do we understand why this is the case?

Reply:
Not yet, but planning to investigate

Reply:
Okay, it's because there are emit-time diagnostics that are not handled properly, the emitted code has then length 0 which causes the "bad IL format". I will fix this.